make_statistic_plots.py

```diff
@@ -0,0 +1,582 @@
+#Author: Nicole Schanche
+#Updated Oct 12, 2022
+#This script makes the TESS statistics plots for tess-website-lite. This is very much a work in progress
+
+import pandas as pd
+import sqlite3
+import json
+import re
+import numpy as np
+#import geograpy
+import itertools
+import plotly.graph_objects as go
+import plotly.express as px
+from datetime import date, datetime
+from dateutil import rrule
+import matplotlib.pyplot as plt
+from wordcloud import WordCloud
+from collections import defaultdict
+from collections import Counter
+
+
+# Get all of the data you will need to make the TESS statistics plots later on
+
+# Read in the latest database file from tpub (Note that this is static for now, but can be adjusted later when we decide how we want to get tpub data for the website. 
+query = sqlite3.connect('content/data/tpub.db').execute('SELECT * FROM pubs')
+
+cols = [column[0] for column in query. description]
+results = pd.DataFrame.from_records(data = query.fetchall(), columns = cols)
+results = results[results['mission'] == 'tess'].reset_index(drop=True)
+
+
+# We will want to pull in some information on the author's affiliation (country and state (if applicable))
+# CSV from Christina
+countries = pd.read_csv("content/data/countries_codes_and_coordinates.csv")
+for column in countries.columns[1:]:
+    countries[column] = [c.strip().replace('"', "") for c in countries[column]]
+countries['Country'] = countries['Country'].replace('Jersey','Bailiwick of Jersey') #Getting confused with New Jersey
+countries.head()
+
+states = (
+    pd.read_csv("content/data/states.csv")
+    .sort_values("State")
+    .reset_index(drop=True)
+)
+states.head()
+
+###################################################################
+#                Interactive journal count Plot
+###################################################################
+
+# Makes a plot of the number of refereed papers over time, broken up by tpub category
+
+ #restrict this plot to entries that are journal articles (vs book chapter, eptrints, conference proceeding, etc)
+is_article = [json.loads(results['metrics'][x])['doctype'] == 'article' for x in range(len(results))]
+article = results[is_article]
+
+article['month'] = pd.to_datetime(article['month'], format="%Y-%m", errors='coerce')
+
+start_date = datetime(2018, 4, 1) #Month of first TESS paper
+end_date = max(article['month']) 
+
+months = rrule.rrule(rrule.MONTHLY, dtstart=start_date, until=end_date) 
+
+labels = ['exoplanets','astrophysics']
```

Comment:
Let's run a linter like [`black`](https://black.readthedocs.io/en/stable/?badge=stable) on these files so that they are formatted nicely and obeys conventions:

Reply:
Done